### PR TITLE
MAT-6106: Add check and maxlength for 64 characters

### PR DESCRIPTION
**Pull request by ethankaplan**
## MADiE PR

Jira Ticket: [MAT-6106](https://jira.cms.gov/browse/MAT-6106)
(Optional) Related Tickets:
https://github.com/MeasureAuthoringTool/madie-cql-library/pull/97
https://github.com/MeasureAuthoringTool/madie-measure/pull/433

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included
* [ ] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is into the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/NewMeasure/CreateNewMeasureDialog.tsx [hunk 1]
```diff
@@ -138,6 +139,23 @@ const CreateNewMeasureDialog = ({ open, onClose }) => {
...
+        "Measure Library name must start with an upper case letter, followed by alpha-numeric character(s) and must not contain spaces or other special characters."
+      );
+    } else if (formik.values.cqlLibraryName.length > 64) {
+      setCqlLibraryHelper(
+        "A Measure Libary Name cannot be more than 64 characters"
+      );
+    } else {
+      setCqlLibraryHelper(null);
+    }
+  }, [formik.values.cqlLibraryName, formik.touched, focusedField]);
```
Comment (adongare) [inline src/components/NewMeasure/CreateNewMeasureDialog.tsx]: could this be moved to validator schema?

File: src/models/MeasureSchemaValidator.ts [hunk 2]
```diff
@@ -20,7 +20,8 @@ export const MeasureSchemaValidator = Yup.object().shape({
     .matches(
       /^[A-Z][a-zA-Z0-9]*$/,
       "Measure Library name must start with an upper case letter, followed by alpha-numeric character(s) and must not contain spaces or other special characters."
-    ),
+    )
+    .max(64, "Measure Library name cannot be more than 64 characters."),
```
Comment (adongare) [inline src/models/MeasureSchemaValidator.ts]: As per ACs, message would be: **Library name cannot be more than 64 characters.**

File: src/components/NewMeasure/CreateNewMeasureDialog.tsx (hunk 1 above)
Comment (ethankaplan) [reply src/components/NewMeasure/CreateNewMeasureDialog.tsx]: done!

File: src/components/NewMeasure/CreateNewMeasureDialog.tsx [hunk 3]
```diff
@@ -247,12 +247,12 @@ const CreateNewMeasureDialog = ({ open, onClose }) => {
             "data-testid": "cql-library-name-input",
             "aria-describedby": "cqlLibraryName-helper-text",
             required: true,
+            maxLength: 64,
           }}
           helperText={
             (formik.touched["cqlLibraryName"] ||
               focusedField === "cqlLibraryName") &&
-            (formikErrorHandler("cqlLibraryName", true) ||
-              "Measure Library name must start with an upper case letter, followed by alpha-numeric character(s) and must not contain spaces or other special characters.")
```
Comment (adongare) [inline src/components/NewMeasure/CreateNewMeasureDialog.tsx]: @mcmcphillips is this related to 508? just want to confirm removing this would not cause a new issue.

Comment (ethankaplan) [reply src/components/NewMeasure/CreateNewMeasureDialog.tsx]: I misinterpreted this code's purpose, added it back

**Review by mcmcphillips — APPROVED** [review]

File: src/components/NewMeasure/CreateNewMeasureDialog.tsx (hunk 1 above)
Comment (mcmcphillips) [reply src/components/NewMeasure/CreateNewMeasureDialog.tsx]: I don't think that code got added to the PR

File: src/models/MeasureSchemaValidator.ts (hunk 2 above)
Comment (mcmcphillips) [reply src/models/MeasureSchemaValidator.ts]: I don't think this was resolved

**Review by mcmcphillips — DISMISSED** [review]
unresolved comments

**Review by adongare — APPROVED** [review]

File: src/components/NewMeasure/CreateNewMeasureDialog.tsx (hunk 1 above)
Comment (ethankaplan) [reply src/components/NewMeasure/CreateNewMeasureDialog.tsx]: That code was removed because it was already covered, my understanding of validations were wrong when I wrote it

**Review by nmorasb — APPROVED** [review]

File: src/models/MeasureSchemaValidator.ts (hunk 2 above)
Comment (ethankaplan) [reply src/models/MeasureSchemaValidator.ts]: This was corrected to what the AC says

**Review by jkotanchik-SB — APPROVED** [review]
